generators/firechip/core/src/main/cc/bridges/cospike/mem_pool.cc

```diff
@@ -0,0 +1,79 @@
+#include "mem_pool.h"
+#include <assert.h>
```

Comment:
What is this `mem_pool`/`thread_pool` stuff for? And why is it under `cospike`? If this was how it was done in fs, then its fine

Reply:
@joonho3020 added this. This was directly copied from FireSim.

Reply:
IIRC, I added this as a performance optimization for the cospike bridge.

Reply:
Ah ok, I wasn't aware of this.